var/spack/repos/builtin/packages/libpeas/package.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+import os
+
+
+class Libpeas(AutotoolsPackage):
+    """libpeas is a gobject-based plugins engine, and is targeted at
+    giving every application the chance to assume its own
+    extensibility."""
+
+    homepage = "http://developer.gnome.org/libpeas/stable"
+
+    version('1.22.0',
+            url='https://download.gnome.org/sources/libpeas/1.22/libpeas-1.22.0.tar.xz',
+            sha256='5b2fc0f53962b25bca131a5ec0139e6fef8e254481b6e777975f7a1d2702a962')
+
+    depends_on('m4', type='build')
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('gettext', type='build')
+    depends_on('pkg-config', type='build')
+    depends_on('atk')
+    depends_on('intltool@0.40.0:')
+    depends_on('xmlto')
+    depends_on('perl')
+    depends_on('perl-xml-parser')
+    depends_on('glib@2.10:')
+    depends_on('gobject-introspection')
+    depends_on('libffi')
+    depends_on('gtkplus')
+    depends_on('gdk-pixbuf')
+    depends_on('pango')
+    depends_on('gnome-common')
+    depends_on('py-pygobject@3:')
+    depends_on('python@3:')
+
+    def setup_environment(self, spack_env, run_env):
+        # Let
+        #
+        # python = self.spec['python']
+        # prefix = python.prefix
+        # pyversion = python.version.up_to(2)
+        # python_lib_path = os.path.join(prefix, 'Frameworks',
+        #                                'Python.framework', 'Versions',
+        #                                pyversion)
+        #
+        # self.spec['python'].libs.ld_flags returns (on macOS)
+        # '-L{0} -lPython'.format(python_lib_path)
+        #
+        # e.g., for python@3.7.4 on macOS via Homebrew, python_lib_path is
+        # /usr/local/opt/python/Frameworks/Python.framework/Versions/3.7
+        #
+        # This directory is correct for many purposes, but libpeas uses the
+        # link flag '-lpython{0}m'.format(pyversion) and does not use an
+        # appropriate -L flag to locate this library, so the correct -L flag
+        # must be appended to LDFLAGS. Furthermore, this library is not found
+        # in python_lib_path.  However, pkg-config returns the correct
```

Comment:
Can you elaborate on "this library is not found in python_lib_path"? Do we need to fix how `spec['python'].libs` works?

Reply:
@adamjstewart No, I don't think `spec['python'].libs` needs to be fixed -- I think it finds the correct library if the build system is set up correctly, though I suppose what's "correct" depends on convention.

To recap the comment, `self.spec['python'].prefix.libs.ld_flags` returns on my macOS `-L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7 -lPython`,
which is the correct library...

...unless a package's build system hardcodes the flag `-lpython3.7m`, which is in `-L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib`, and is symlinked to the `Python` library. The `libpeas` build system also omits `-L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib`, so I have to add this `-L` flag.

The command `pkg-config --libs self.spec['python'].prefix.lib.pkgconfig` returns `-L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib -lpython3.7m`. The latter flag is redundant, but this solution seems like it will generalize correctly.